tests/getdel_test.go

```diff
@@ -0,0 +1,59 @@
+package tests
+
+import (
+	"gotest.tools/v3/assert"
+	"testing"
+	"time"
+)
+
+func TestGetDel(t *testing.T) {
+	conn := getLocalConnection()
+	defer conn.Close()
+
+	testCases := []struct {
+		name   string
+		cmds   []string
+		expect []interface{}
+		delays []time.Duration
+	}{
+		{
+			name:   "GetDel",
+			cmds:   []string{"SET k v", "GETDEL k", "GETDEL k", "GET k"},
+			expect: []interface{}{"OK", "v", "(nil)", "(nil)"},
+			delays: []time.Duration{0, 0, 0, 0},
+		},
+		{
+			name:   "GetDel with expiration, checking if key exist and is already expired, then it should return null",
+			cmds:   []string{"GETDEL k", "SET k v EX 4", "GETDEL k"},
+			expect: []interface{}{"(nil)", "OK", "(nil)"},
+			delays: []time.Duration{0, 0, 5 * time.Second},
+		},
+		{
+			name: "GetDel with expiration, checking if key exist and is not yet expired, then it should return its " +
+				"value",
+			cmds:   []string{"SET k v EX 40", "GETDEL k"},
+			expect: []interface{}{"OK", "v"},
+			delays: []time.Duration{0, 5 * time.Second},
+		},
+		{
+			name: "GetDel with invalid command",
+			cmds: []string{"GETDEL", "GETDEL k v"},
+			expect: []interface{}{"ERR wrong number of arguments for 'getdel' command",
+				"ERR wrong number of arguments for 'getdel' command"},
+			delays: []time.Duration{0, 0},
```

Comment:
Please use shorter expiration times and delays (1-2 seconds), this will help the test suite run faster.

Reply:
Done